### DISCUSSION TITLE
Image fields not returned properly when translated by WPML

### DISCUSSION THREAD

**Discussion by Shelob9** [pods-framework/pods#6590] [Ideas]
In [this forum thread](http://pods.io/forums/topic/problem-with-displaying-images-from-pods/) a user is not able to get values from a Pods image field in the translated version of their site. 

In [this response](http://pods.io/forums/topic/problem-with-displaying-images-from-pods/#post-180693) they show that the image is not being stored properly in wp_podsrel for the translated field.

**Comment by zigotica** [comment]
hi, any news on this issue? i'm having the exact same problem, aparently we're not alone:
https://wpml.org/forums/topic/displaying-images-from-pods-field-field-type-file-image-video-in-transla/

thanx!

**Comment by sc0ttkclark** [comment]
@Shelob9 is looking into this

**Comment by zigotica** [comment]
THANX!

**Comment by Shelob9** [comment]
I've started getting up to speed with WPML and tested this issue. In a custom post type I have created an image field  called 'screenshots'. In the translation options for the post in the CPT, I see options for both _pods_screenshots and screenshots. No matter what combination of settings I use for these two fields, nothing is returned when translating the post.

I have a Pods template that contains `<img src={@screenshot} />` and it works as expected when not translated, but when not translated it returns `<img src=>`

I also tried using the pods class to get the value of the field. `$screenshot = $obj->field('screenshots' );$out = var_dump($screenshot);` produced the array you would expect when not translated, and when translated the output was 'bool(false)'.

**Comment by Shelob9** [comment]
@zigotica Is there anyway I can review how you have WPML setup with Pods on your site? it would be very helpful for my efforts in documenting and testing WPML + Pods to see how you're using it and could help me get to the bottom of this issue more quickly.

If possible, I'd love to be able to login to your site and see how you set things up, I promise not to change anything. If you're OK with that you can submit login credentials securely here: pods.io/help

Please reference this issue and mention the credentials are for Josh.

**Comment by zigotica** [comment]
Hey @Shelob9 thanx for taking care of the issue!

Unfortunately, our site is in a private network, there's no way i can grant you access, sorry. But i think you don't need it, since a default install of both WPML+Pods will show you the problem, as you have previously indicated. If you really need to test in another server/options we can see if it would be possible to install and replicate the conflictive pod and field into a custom setup open for you.

S

**Comment by Shelob9** [comment]
@zigotica No worries.

I tested this further and even more confused.

I added additional posts in my CPT. Whenever I duplicate the post to created the spanish version  the image field has the same image, which is the first image that WPML ever duplicated, no matter what image was set in that image field. That image does return properly.

So it seems like the problem may be in how WPML copies the value of the media field.

**Comment by zigotica** [comment]
ok, thanx anyway for trying, @Shelob9

**Comment by Shelob9** [comment]
@zigotica I am actively working with WPML on this and will keep you up to date.

**Comment by zigotica** [comment]
wow THANK YOU so much, @Shelob9 :)

**Comment by pglewis** [comment]
There is much to consider here and there are related issues such as #1851.  I'm currently banging heads together with @unknownnf on this as well as the broader scope.

**Comment by unknownnf** [comment]
I've been trying to reproduce this without success. Anyone can shed some info on the issue, some step by step guide to break it?

**Comment by robinpiets** [comment]
Hi guys.

Is there any progress on this bug?

We am having the same issue, but it started when we installed WPML Media. Now we can't edit English posts, because we're losing all the images in the 'Pods Image/Media field'.

I noticed that if I change the GET variable 'lang' in the URL from 'en' (English) to 'nl' (Dutch) that the image is showing up.

For example: site.com/wp-admin/post.php?post=14093&action=edit&lang=en
Changing to: site.com/wp-admin/post.php?post=14093&action=edit&lang=nl

We're using Wordpress 3.9.1 and this plugins:
- Pods - Custom Content Types and Fields 2.4
- WPML Media 2.1.4
- WPML Multilingual CMS 3.1.5 (default language: Dutch)

Hope someone finds a solution for this soon... Thanks for looking into it!

Robin

**Comment by robinpiets** [comment]
Hi again.

I think I have found the clue for this bug. At least this seems to be the problem at my setup.

I have a Pods Media Field in my Posts and I have two languages (Dutch and English).

I first create a Dutch post and add media (Dutch) to the Pods Media Field and I hit Publish everything works fine. The media appears in the Admin and on the site. But, when I translate this post to English (using the change language flags in the Admin bar) and I add the same media files to the Pods Media Field and hit Publish, than the images don't appear on the Admin page, but they do appear on the Site.

It looks like the Dutch media files were added to the English post and that's why they don't show up. In my last posts I said that when I change the  "GET variable 'lang' in the URL from 'en' (English) to 'nl' (Dutch) that the image is showing up" in the Admin. But I have to re-add the images in the English post to really attach them.

That seems to be a bug in WPML or WPML Media which I would love to see fixed. Now I have to tell the translators that they have to Translate a post and add some data to it, then save the draft (so it turns into English) and then add media to it. That's a little bit silly, right?

@sc0ttkclark Shall I put this message in the WPML Support?

Thanks a million for the best plugins for Wordpress! Hope to hear from you soon.

Robin

**Comment by Shelob9** [comment]
@Griptaped Thanks for this information. We will be looking more into fixing multi-language issues in pods 3.0 and this helps. No need to report to WPML. We will get int ouch with them if we feel like this is something they need to fix on there end opposed to being something we need to fix in Pods.

**Comment by pglewis** [comment]
I know this continues get kicked down the road, but it's out of scope of the 3.0 milestone focus.  Sure willing to consider it for 3.0 if a contributor parachutes in.

**Comment by JoryHogeveen** [comment]
I believe this is fixed in Pods 2.6.6 (#3574)

EDIT: Nevermind, related to #1851 see my comment in this issue for Polyland (9 aug 2016), I think similar functionality will need to be build for WPML

**Comment by quasel** [comment]
@JoryHogeveen another one - I guess this is still an issue?
